superset-frontend/src/dashboard/components/EmbeddedModal/EmbeddedModal.test.tsx

```diff
@@ -0,0 +1,187 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { fireEvent, waitFor } from '@testing-library/react';
+import { render, screen } from 'spec/helpers/testing-library';
+import '@testing-library/jest-dom';
+import {
+  SupersetApiError,
+  getExtensionsRegistry,
+  makeApi,
+} from '@superset-ui/core';
+import setupExtensions from 'src/setup/setupExtensions';
+import DashboardEmbedModal from './index';
+
+const defaultResponse = {
+  result: { uuid: 'uuid', dashboard_id: '1', allowed_domains: ['example.com'] },
+};
+
+jest.mock('@superset-ui/core', () => ({
+  ...jest.requireActual<any>('@superset-ui/core'),
+  makeApi: jest.fn(),
+}));
+
+const mockOnHide = jest.fn();
+const defaultProps = {
+  dashboardId: '1',
+  show: true,
+  onHide: mockOnHide,
+};
+const resetMockApi = () => {
+  (makeApi as any).mockReturnValue(
+    jest.fn().mockResolvedValue(defaultResponse),
+  );
+};
+const setMockApiNotFound = () => {
+  const notFound = new SupersetApiError({ message: 'Not found', status: 404 });
+  (makeApi as any).mockReturnValue(jest.fn().mockRejectedValue(notFound));
+};
+
+const setup = () => {
+  render(<DashboardEmbedModal {...defaultProps} />, { useRedux: true });
+  resetMockApi();
+};
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  resetMockApi();
+});
+
+it('renders', async () => {
+  setup();
+  await waitFor(() => {
+    expect(screen.getByText('Embed')).toBeInTheDocument();
+  });
+});
+
+it('displays loading state', async () => {
+  setup();
+  await waitFor(() => {
```

Comment:
You can use `findBy` to make it simpler. Generally, `waitFor` is useful when you need to use `not`.